### ISSUE TITLE
Enhance assertion messages in base.py for more informative error reporting

### ISSUE THREAD

**Issue by cmower** [dfki-ric/movement_primitives#60]
The assertion messages in the following could be more useful

https://github.com/dfki-ric/movement_primitives/blob/d187c1598c8a429df4c546b2a98994c0a468b80d/movement_primitives/base.py#L107-L123

For example, instead of 

```
    assert len(start_y) == self.n_dims, f"{len(start_y)}"
AssertionError: 6
```

they would say 

```
    assert len(start_y) == self.n_dims, f"{len(start_y)}"
AssertionError: 6 elements appear in start_y, expected 5
```

If you agree, I'd be happy to setup a PR?

**Comment by AlexanderFabisch** [comment]
Hi, I'd rather raise ValueErrors in that case. This is just a quick and dirty hack at the moment! A PR would be nice. :)

**Comment by AlexanderFabisch** [comment]
Closed with #61
